### PR TITLE
validation corretion

### PR DESCRIPTION
**Pull request by JihongJeong**
Divided train and validation set before running epochs to make validation set is independent to epochs

(for the change of freqencoder.py, I'm steel working on it. so you don't have to look at it right now)

### CODE REVIEW

**Comment by justin4ai** [comment]
Thanks for your hard work. Let me seee

File: train.py [hunk 1]
```diff
@@ -40,54 +44,51 @@ def main(args):
         train_loss = None
         validation_acc = 0
         validation_loss = None
-        split_val = int(len(dataloader) * 0.8)
-        for batch_num, batch in enumerate(tqdm(dataloader)):
-            if batch_num < split_val:
-                optimizer.zero_grad()
-                
-                img, labels = batch
+        for train_idx, train_batch in enumerate(tqdm(train_dataloader)):
```
Comment (justin4ai) [inline train.py]:
```suggestion
        for train_idx, train_batch in enumerate(train_dataloader):
```

I don't think tqdm is needed since we print the log out at each 10th iteration while tqdm does that at each iteration, therefore they collide.

File: train.py [hunk 2]
```diff
@@ -40,54 +44,51 @@ def main(args):
...
+                if labels[idx] == logits[idx]:
+                    train_acc = train_acc + 1
+            
+            if train_loss is None:
+                train_loss = loss
+            else:
+                train_loss = torch.mean(torch.stack([train_loss, loss]))
+            
+            if train_idx%10 == 0:
+                print(f"\nTraining accuracy : {train_acc}/{(train_idx + 1) * batch_size},\tTrining Loss : {train_loss}")
```
Comment (justin4ai) [inline train.py]:
```suggestion
                print(f"\n{train_idx}/{len(train_dataloader)}th iteration : Training accuracy : {(train_acc/((train_idx + 1) * batch_size))*100:.2f}%,\tTraining Loss : {train_loss}")
```

This print-out line will be fancier.

**Review by justin4ai — APPROVED** [review]
I made(committed) two lines of modification for better log printing.

Also I don't think we need to do validation on all the 20% for each epoch - probably checking just one random mini batch from it at each epoch could be enough. You know this suggestion is not that high-priority though.

Thanks for your hard work. I merge this to detector.

*Please manually apply two lines of modification I made to yours as well if you already have started working with your detector-tmp02 branch after sending this PR.

**Comment by justin4ai** [comment]
Sorry but please replace the print line I've committed with the following line:

```python
print(f"\nEpoch {epoch} - {train_idx}/{len(train_dataloader)}th iteration : Training accuracy : {(train_acc/((train_idx + 1) * batch_size))*100:.2f}%,\tTraining Loss : {train_loss}")
``` 

where epoch is printed as well.
